### PR TITLE
[SPARK-23691][PYTHON] Use sql_conf util in PySpark tests where possible

### PR DESCRIPTION
**Pull request by HyukjinKwon**
## What changes were proposed in this pull request?

https://github.com/apache/spark/commit/d6632d185e147fcbe6724545488ad80dce20277e added an useful util

```python
@contextmanager
def sql_conf(self, pairs):
    ...
```

to allow configuration set/unset within a block:

```python
with self.sql_conf({"spark.blah.blah.blah", "blah"})
    # test codes
```

This PR proposes to use this util where possible in PySpark tests.

Note that there look already few places affecting tests without restoring the original value back in unittest classes.

## How was this patch tested?

Manually tested via:

```
./run-tests --modules=pyspark-sql --python-executables=python2
./run-tests --modules=pyspark-sql --python-executables=python3
```

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
@ueshin and @BryanCutler, could you take a look when you are available?

**Comment by HyukjinKwon** [comment]
BTW, I double checked it produces the stack trace fine by manually changing some tests locally.

File: python/pyspark/sql/tests.py
Comment (ueshin) [inline python/pyspark/sql/tests.py]: We can omit this when we use the default value or set the value in setup method, but I'm okay if we want to show the value explicitly.

**Review by ueshin — APPROVED** [review]
LGTM except for one comment, but I'd leave it to @HyukjinKwon whether we omit the default value block or not.

File: python/pyspark/sql/tests.py
Comment (BryanCutler) [inline python/pyspark/sql/tests.py]: So the `sql_conf` context will change this back to be unset right?

**Review by BryanCutler — APPROVED** [review]
Thanks @HyukjinKwon , LGTM!

File: python/pyspark/sql/tests.py
Comment (HyukjinKwon) [reply python/pyspark/sql/tests.py]: Yup, it originally unset `spark.sql.crossJoin.enabled` but now it set to the original value back.
If `spark.sql.crossJoin.enabled` is unset in this test, it will change this back to be unset.

File: python/pyspark/sql/tests.py
Comment (HyukjinKwon) [reply python/pyspark/sql/tests.py]: Ah, OK. I am fine. will omit this.

**Review by felixcheung — APPROVED** [review]
nice!

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by HyukjinKwon** [comment]
@BryanCutler, could I have the very first PR merged by you as a new fresh committer :-)?

I personally think it might be good to merge to branch-2.3 if it doesn't have conflicts. If it has, I think we are fine to get this into master only for now.

**Comment by BryanCutler** [comment]
Sure thing! I probably won't be able to until later tonight, but I'll give
it a shot as soon as I can, hopefully it will work :-D

On Mar 18, 2018 10:24 PM, "Hyukjin Kwon" <notifications@github.com> wrote:

@BryanCutler <https://github.com/bryancutler>, could I have the very first
PR merged by you as a new fresh committer :-)?

I personally think it might be good to merge to branch-2.3 if it doesn't
have conflicts. If it has, I think we are fine to get this into master only
for now.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/apache/spark/pull/20830#issuecomment-374104603>, or mute
the thread

**Review by dongjoon-hyun — APPROVED** [review]

**Comment by BryanCutler** [comment]
Merged to master!  (I think it went ok..)  Thanks @HyukjinKwon !!

**Comment by HyukjinKwon** [comment]
Thanks for reviewing and merging this @ueshin, @felixcheung, @BryanCutler and @dongjoon-hyun.

(Just FYI, I usually manually resolve JIRAs when I accidentally failed to take an action with the merge script. I think that's fine.)

**Comment by BryanCutler** [comment]
The cherry pick to branch-2.3 did have some conflicts.  Just to check for the reason to backport, even though this isn't a bug it's pretty safe and will help keep things inline so less conflicts for future backports?

**Comment by HyukjinKwon** [comment]
Yup, that was exactly what I thought. I think it's fine to not bother backport too since it has conflicts.

**Comment by HyukjinKwon** [comment]
But I am willing to do this if you think it's better to do this. No objection.

**Comment by BryanCutler** [comment]
Hmm, it looks like the conflict is just one block with group agg tests, probably not a big deal - you want to take a look?

**Comment by HyukjinKwon** [comment]
Sure, will open a PR soon.

**Comment by HyukjinKwon** [comment]
Ahhhh .. https://github.com/apache/spark/commit/d6632d185e147fcbe6724545488ad80dce20277e added the util into master only ...

**Comment by HyukjinKwon** [comment]
Let me open a PR and cc you guys to show the diff.

**Pull request closed without merging**
